tests/integration-tests/conftest.py

```diff
@@ -608,6 +609,12 @@ def inject_additional_config_settings(  # noqa: C901
     with open(cluster_config, encoding="utf-8") as conf_file:
         config_content = yaml.safe_load(conf_file)
 
+    if not dict_has_nested_key(config_content, ("HeadNode", "Ssh", "AllowedIps")):
+        # If the test is running in an EC2 instance limit SSH connection access from instance running the test
+        instance_public_ip = get_metadata("public-ipv4", raise_error=False)
+        if instance_public_ip:
+            dict_add_nested_key(config_content, f"{instance_public_ip}/32", ("HeadNode", "Ssh", "AllowedIps"))
```

Comment:
Are we sure that public ips are always assigned to the calling instance?
If not, we should use private IP that we know is always assigned, right?